Making jemalloc an optional feature

SebastianSchildt opened the pull request:
Should solve #460

argerus reviewed (changes requested):
Looks good, .
But can I suggest that we call the feature `jemalloc` instead.

argerus commented on kuksa_databroker/databroker/Cargo.toml:
```diff
@@ -48,8 +48,8 @@ sqlparser = "0.16.0"
-version = "0.5.0"
+# to enable jemmalloc use --features jemallocator
+jemallocator = { version =  "0.5.0", optional = true }
 
```
```suggestion
[features]
jemalloc = ["dep:jemallocator"]

```

SebastianSchildt commented:
Fixed.
Probably not directly related, but my cargo was thinking serde_derive was missing from Cargo.lock, so I also added it, so at least version is pinned.
